src/sentry/sentry_metrics/client/snuba.py

```diff
@@ -0,0 +1,224 @@
+        project_id: int,
+        metric_name: str,
+        value: Union[int, float],
+        tags: Mapping[str, str],
+        unit: Optional[str],
+    ) -> None:
+
+        """
+        Emit a counter metric for internal use cases only.
+        Note that, as of now, this function will return
+        immediately even if the metric message has not been
+        produced to the broker yet.
+        """
+
+        # the message that the indexer receives
+        counter_metric = {
+            "org_id": org_id,
+            "project_id": project_id,
+            "name": build_mri(metric_name, "c", use_case_id, unit),
+            "value": value,
+            "timestamp": int(datetime.now().timestamp()),
+            "tags": tags,
+            "retention_days": get_retention_from_org_id(org_id),
+            "type": "c",
+        }
+
+        self.__build_and_send_request(counter_metric)
+
+    def set(
+        self,
+        use_case_id: UseCaseID,
+        org_id: int,
+        project_id: int,
+        metric_name: str,
+        value: Sequence[int],
+        tags: Mapping[str, str],
+        unit: Optional[str],
+    ) -> None:
+
+        """
+        Emit a set metric for internal use cases only. Can support
+        a sequence of values. Note that, as of now, this function
+        will return immediately even if the metric message has not been
+        produced to the broker yet.
+        """
+
+        set_metric = {
+            "org_id": org_id,
+            "project_id": project_id,
+            "name": build_mri(metric_name, "s", use_case_id, unit),
+            "value": value,
+            "timestamp": int(datetime.now().timestamp()),
+            "tags": tags,
+            "retention_days": get_retention_from_org_id(org_id),
+            "type": "s",
+        }
+
+        self.__build_and_send_request(set_metric)
+
+    def distribution(
+        self,
+        use_case_id: UseCaseID,
+        org_id: int,
+        project_id: int,
+        metric_name: str,
+        value: Sequence[Union[int, float]],
+        tags: Mapping[str, str],
+        unit: Optional[str],
+    ) -> None:
+
+        """
+        Emit a distribution metric for internal use cases only. Can
+        support a sequence of values. Note that, as of now, this function
+        will return immediately even if the metric message has not been
+        produced to the broker yet.
+        """
+        dist_metric = {
+            "org_id": org_id,
+            "project_id": project_id,
+            "name": build_mri(metric_name, "d", use_case_id, unit),
+            "value": value,
+            "timestamp": int(datetime.now().timestamp()),
+            "tags": tags,
+            "retention_days": get_retention_from_org_id(org_id),
+            "type": "d",
+        }
+
+        self.__build_and_send_request(dist_metric)
+
+    def __build_payload(self, metric) -> Sequence[Any]:
+        # build message batch for the MessageProcessor
+        message_batch = [
```

Comment:
*nit:* constructing kafka-specific things like this is pretty gross fwiw but i think it is still much better than duplicating biz logic

ideally we'd clean up the interface for MessageProcessor so that this is not necessary (def not as part of this PR)